src/com/owncloud/android/lib/resources/files/ChunkedUploadRemoteFileOperation.java

```diff
@@ -162,7 +162,11 @@ protected int uploadFile(OwnCloudClient client) throws IOException {
 
         } finally {
             SharedPreferences.Editor editor = sharedPref.edit();
-            editor.putStringSet(chunkId, successfulChunks).apply();
+            if (this.isSuccess(status)) {
```

Comment:
the success part seems to be exactly the same as the last lines of the try {} part and the finally block will always be executed.